src/main/java/net/shadowmage/ancientwarfare/npc/entity/faction/NpcFactionSpellcasterWizardry.java

```diff
@@ -0,0 +1,182 @@
+	private Spell continuousSpell;
+	private int spellCounter;
+
+	@SuppressWarnings("unused")
+	public NpcFactionSpellcasterWizardry(World world) {
+		super(world);
+		addAI();
+		tasks.addTask(3, spellCastingAI);
+	}
+
+	@SuppressWarnings("unused")
+	public NpcFactionSpellcasterWizardry(World world, String factionName) {
+		super(world, factionName);
+		addAI();
+		tasks.addTask(3, spellCastingAI);
+	}
+
+	@SuppressWarnings("Duplicates")
+	private void addAI() {
+		tasks.addTask(0, new EntityAISwimming(this));
+		tasks.addTask(0, new EntityAIRestrictOpenDoor(this));
+		tasks.addTask(0, new NpcAIDoor(this, true));
+		tasks.addTask(1, new NpcAIFollowPlayer(this));
+		tasks.addTask(2, new NpcAIMoveHome(this, 50.f, 5.f, 30.f, 5.f));
+		tasks.addTask(2, new NpcAIFactionArcherStayAtHome(this));
+
+		tasks.addTask(101, new EntityAIWatchClosest2(this, EntityPlayer.class, 3.0F, 1.0F));
+		tasks.addTask(102, new NpcAIWander(this));
+		tasks.addTask(103, new NpcAIWatchClosest(this, EntityLiving.class, 8.0F));
+
+		targetTasks.addTask(1, new NpcAIHurt(this));
+		targetTasks.addTask(2, new NpcAIAttackNearest(this, this::isHostileTowards));
+	}
+
+	public void setSpells(List<Spell> spells) {
+		this.spells = spells;
+	}
+
+	@Override
+	public List<Spell> getSpells() {
+		return spells;
+	}
+
+	@Override
+	public void setSpellCounter(int count) {
+		spellCounter = count;
+	}
+
+	@Override
+	public int getSpellCounter() {
+		return spellCounter;
+	}
+
+	@Override
+	public SpellModifiers getModifiers() {
+		SpellModifiers spellModifiers = new SpellModifiers();
+		spellModifiers.set("potency", 1.5f, true);
+		return spellModifiers;
+	}
+
+	@Override
+	public void setContinuousSpell(Spell spell) {
+		continuousSpell = spell;
+	}
+
+	@Override
+	public Spell getContinuousSpell() { return continuousSpell; }
+
+	@Override
+	public int getAimingError(EnumDifficulty difficulty) { // for spells
+		switch (difficulty) {
+			case EASY:
+				return 7;
+			case NORMAL:
+				return 4;
+			case HARD:
+				return 1;
+			default:
+				return 7; // Peaceful counts as easy
+		}
+	}
+
+	@Override
+	public boolean hasAltGui() {
+		return true;
+	}
+
+	@Override
+	public void openAltGui(EntityPlayer player) {
+		NetworkHandler.INSTANCE.openGui(player, NetworkHandler.GUI_NPC_FACTION_SPELLCASTER_WIZARDRY, getEntityId(), 0, 0);
+	}
+
+	@Override
+	public void writeEntityToNBT(NBTTagCompound nbt) {
+		super.writeEntityToNBT(nbt);
+		nbt.setTag("spells", NBTExtras.listToNBT(getSpells(), spell -> new NBTTagString(spell.getRegistryName().toString())));
+	}
+
+	@Override
+	public void writeAdditionalItemData(NBTTagCompound nbt) {
+		nbt.setTag("spells", NBTExtras.listToNBT(getSpells(), spell -> new NBTTagString(spell.getRegistryName().toString())));
+		super.writeAdditionalItemData(nbt);
+	}
+
+	@Override
+	public void readEntityFromNBT(NBTTagCompound nbt) {
+		super.readEntityFromNBT(nbt);
+		spells = ((List<Spell>) NBTExtras.NBTToList(nbt.getTagList("spells", Constants.NBT.TAG_STRING),
```

Comment:
NBTExtras.NBTToList should be replaced with call to NBTHelper, we don't have a list one yet but you can just recreate what is there already with getSet and create a new getList method that instead of creating a hashset creates arraylist